lib/internal/process/promises.js

```diff
@@ -205,6 +218,23 @@ function processPromiseRejections() {
         emitUnhandledRejectionWarning(uid, reason);
         break;
       }
+      case kThrowUnhandledRejections: {
+        const handled = process.emit('unhandledRejection', reason, promise);
+        if (!handled) {
+          const err = reason instanceof Error ?
```

Comment:
what happens if it's an Error from the vm module? (where `instanceof` won't work, but it's still an error with a stack trace)

Reply:
That's an interesting point! If you scroll up, you'll see that I just copied and pasted this code from the implementation for `--strict`, which I'd assumed was adequately vetted.

I'm open to improving the implementation here, if you have a recommendation.

Reply:
Considering `util.isError` wrongly passes on `{ __proto__: Error.prototype }`, and there's no way in the spec to robustly determine errors, and this is already the approach used above, we might as well keep it for now - but hopefully a future PR can fix this.

Reply:
We also have https://github.com/nodejs/node/blob/54b36e401d2b72d95e5f1dbbc787f6beed639347/lib/internal/util.js#L51-L56 It'll detect errors from other realms while still being lenient about `{ __proto__: Error.prototype }`.

Reply:
I don't really know how to think about or test errors from other realms. If there's a bug here, I don't know how to repro it, and regardless, none of that is really my focus/interest in filing this PR; my goal here is just to add the new options while avoiding regressions.

I _believe_ that this code will (regardless) do something pretty reasonable. Even if it fails to identify an Error-ish object as an Error, the code will just convert the error to a string with `noSideEffectsToString`.

I suggest that if y'all would like to see an improvement here, file a separate issue or PR to use `util.isError`, `util.types.isNativeError`, or whatever you think is best.